src/imars3dv2/filters/denoise.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import numpy as np
+import tomopy
+import tomopy.util.mproc as mproc
+import concurrent.futures as cf
+from scipy.signal import convolve2d
+from scipy.ndimage import median_filter
+from skimage.restoration import denoise_bilateral
+
+
+def measure_noiseness(image: np.ndarray) -> float:
+    """
+    Measure the noiseness of the image, doi:10.1006/cviu.1996.0060
+
+    Parameters
+    ----------
+    @param image:
+        The image to measure the noiseness of.
+
+    Returns
+    -------
+    @return:
+        The noiseness of the image.
+    """
+    # rescale to [0, 255]
+    image = (image - image.min()) / (image.max() - image.min()) * 255
+    # compute factor
+    height, width = image.shape
+    factor = np.sqrt(np.pi / 2.0) / (6 * (height - 2) * (width - 2))
+    # set kernel
+    kernel = np.array(
+        [
+            [1, -2, 1],
+            [-2, 4, -2],
+            [1, -2, 1],
+        ]
+    )
+    # compute
+    sigma = np.sum(np.sum(np.absolute(convolve2d(image, kernel))))
+    return factor * sigma
+
+
+def measure_sharpness(image: np.ndarray) -> float:
+    """
+    Measure the sharpness of the image, using norm of gradient as approximation.
+
+    Parameters
+    ----------
+    @param image:
+        The image to measure the sharpness of.
+
+    Returns
+    -------
+    @return:
+        The sharpness of the image.
+    """
+    # rescale to [0, 255]
+    image = (image - image.min()) / (image.max() - image.min()) * 255
+    # compute gradient
+    gx, gy = np.gradient(image)
+    # use norm of gradient as approximation for sharpness
+    return np.sqrt(gx ** 2 + gy ** 2).mean()
```

Comment:
I think
```suggestion
    return np.sqrt(np.square(gx) + np.square(gy)).mean()
```
executes more quickly, but I may be out of date